core/cloudflow-blueprint/src/main/scala/cloudflow/blueprint/Topic.scala

```diff
@@ -115,12 +116,14 @@ final case class Topic(
     if (otherPort.portPath != port.portPath) {
       val schema      = port.schemaDescriptor
       val otherSchema = otherPort.schemaDescriptor
-
+      // TODO make this more dynamic / pluggable
       if (otherSchema.format == schema.format &&
           otherSchema.fingerprint == schema.fingerprint) {
         None
       } else if (otherSchema.format == AvroFormat) {
         checkAvroCompatibility(port, otherPort)
+      } else if (otherSchema.format == ProtoFormat) {
```

Comment:
I think both of the above if statements need to check the schema of both ports

`} else if (otherSchema.format == AvroFormat && port.schemaDescriptor.format == AvroFormat) {
...
} else if (otherSchema.format == ProtoFormat && port.schemaDescriptor.format == ProtoFormat) {`

otherwise, a not useful error is thrown on verifyblueprint in the case that you accidentally pair up a protobuf outlet with an avro inlet or visa versa